sharding-core/sharding-core-preprocessor/src/test/java/org/apache/shardingsphere/core/optimize/segment/select/pagination/engine/TopPaginationContextEngineTest.java

```diff
@@ -0,0 +1,90 @@
+    public void setUp() {
+        topPaginationContextEngine = new TopPaginationContextEngine();
+    }
+    
+    @Test
+    public void assertCreatePaginationContextWhenRowNumberPredicateNotPresent() {
+        TopSegment topSegment = new TopSegment(0, 10, "text", null, "rowNumberAlias");
+        Collection<AndPredicate> andPredicates = Collections.emptyList();
+        List<Object> parameters = Collections.emptyList();
+        PaginationContext paginationContext = topPaginationContextEngine.createPaginationContext(topSegment, andPredicates, parameters);
+        assertThat(paginationContext.getOffsetSegment(), is(Optional.<PaginationValueSegment>absent()));
+        assertThat(paginationContext.getRowCountSegment(), is(Optional.<PaginationValueSegment>absent()));
+    }
+    
+    @Test
+    public void assertCreatePaginationContextWhenRowNumberPredicatePresentAndOperatorIsGreatThan() {
+        String name = "rowNumberAlias";
+        ColumnSegment columnSegment = new ColumnSegment(0, 10, name);
+        PredicateCompareRightValue predicateCompareRightValue = mock(PredicateCompareRightValue.class);
+        when(predicateCompareRightValue.getOperator()).thenReturn(">");
+        when(predicateCompareRightValue.getExpression()).thenReturn(new LiteralExpressionSegment(0, 10, 100));
+        PredicateSegment predicateSegment = new PredicateSegment(0, 10, columnSegment, predicateCompareRightValue);
+        AndPredicate andPredicate = new AndPredicate();
+        andPredicate.getPredicates().add(predicateSegment);
+        Collection<AndPredicate> andPredicates = Collections.singleton(andPredicate);
+        ProjectionsContext projectionsContext = mock(ProjectionsContext.class);
+        String rowNumberAlias = "predicateRowNumberAlias";
+        when(projectionsContext.findAlias(anyString())).thenReturn(Optional.of(rowNumberAlias));
+        List<Object> parameters = Collections.emptyList();
+        TopSegment topSegment = new TopSegment(0, 10, "text", null, name);
+        PaginationContext paginationContext = topPaginationContextEngine.createPaginationContext(topSegment, andPredicates, parameters);
+        Optional<PaginationValueSegment> paginationValueSegmentOptional = paginationContext.getOffsetSegment();
+        assertTrue(paginationValueSegmentOptional.isPresent());
+        PaginationValueSegment paginationValueSegment = paginationValueSegmentOptional.get();
+        assertTrue(paginationValueSegment instanceof NumberLiteralRowNumberValueSegment);
+        NumberLiteralRowNumberValueSegment numberLiteralRowNumberValueSegment = (NumberLiteralRowNumberValueSegment) paginationValueSegment;
+        assertThat(numberLiteralRowNumberValueSegment.getStartIndex(), is(0));
+        assertThat(numberLiteralRowNumberValueSegment.getStopIndex(), is(10));
+        assertThat(numberLiteralRowNumberValueSegment.getValue(), is(100L));
+        assertThat(paginationContext.getRowCountSegment(), is(Optional.<PaginationValueSegment>absent()));
```

Comment:
Can use `assertFalse(paginationContext.getRowCountSegment().isPresent())` to instead of binary assertion.
Using unary assertion for high priority if it passable.